src/OrleansAWSUtils/Streams/SQSAdapterReceiver.cs

```diff
@@ -0,0 +1,119 @@
+﻿using Orleans;
+using Orleans.Runtime;
+using Orleans.Streams;
+using OrleansAWSUtils.Storage;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using SQSMessage = Amazon.SQS.Model.Message;
+
+namespace OrleansAWSUtils.Streams
+{
+    /// <summary>
+    /// Recieves batches of messages from a single partition of a message queue.  
+    /// </summary>
+    internal class SQSAdapterReceiver : IQueueAdapterReceiver
+    {
+        private SQSStorage queue;
+        private long lastReadMessage;
+        private Task outstandingTask;
+        private readonly Logger logger;
+        
+
+        public QueueId Id { get; private set; }
+
+        public static IQueueAdapterReceiver Create(QueueId queueId, string dataConnectionString, string deploymentId)
+        {
+            if (queueId == null) throw new ArgumentNullException("queueId");
+            if (string.IsNullOrEmpty(dataConnectionString)) throw new ArgumentNullException("dataConnectionString");
+            if (string.IsNullOrEmpty(deploymentId)) throw new ArgumentNullException("deploymentId");
+
+            var queue = new SQSStorage(queueId.ToString(), dataConnectionString, deploymentId);
```

Comment:
KK.  If queueId is created with a queue prefix that does not follow there rules, it could break, but this is safe enough.